botocore/regions.py

```diff
@@ -661,59 +661,113 @@ def auth_schemes_to_signing_context(self, auth_schemes):
-            elif 'signingRegionSet' in scheme:
-                # todo: can we handle lists of regions?
-                if len(scheme['signingRegionSet'][0]) > 0:
-                    signing_context['region'] = scheme['signingRegionSet'][0]
-            if 'signingName' in scheme:
-                signing_context.update(signing_name=scheme['signingName'])
-            if 'disableDoubleEncoding' in scheme:
-                signing_context['disableDoubleEncoding'] = ensure_boolean(
-                    scheme['disableDoubleEncoding']
+        if self._requested_auth_scheme is not None:
+            try:
+                # Use the first scheme that matches the requested scheme,
+                # after accounting for naming differences between botocore and
+                # endpoint rulesets. Keep the requested name.
+                name, scheme = next(
+                    (self._requested_auth_scheme, s)
+                    for s in auth_schemes
+                    if self._does_botocore_authname_match_ruleset_authname(
+                        self._requested_auth_scheme, s['name']
+                    )
                 )
-
-            LOG.debug(
-                'Selected auth type "%s" with signing context params: %s',
-                scheme['name'],
-                signing_context,
-            )
-            return scheme['name'], signing_context
+            except StopIteration:
+                # For legacy signers, no match will be found. Do not raise an
+                # exception, instead default to the logic in botocore
+                # customizations.
+                return None, {}
         else:
-            # If an authSchemes list is present in the Endpoint object but none
-            # of the entries are supported, raise an exception. (However, if
-            # authSchemes is not present, use the service's default auth type.)
-            fixable_with_crt = False
-            auth_type_options = [scheme['name'] for scheme in auth_schemes]
-            if not HAS_CRT:
-                fixable_with_crt = any(
-                    scheme in CRT_SUPPORTED_AUTH_TYPES
-                    for scheme in auth_type_options
+            try:
+                name, scheme = next(
+                    (s['name'], s)
+                    for s in auth_schemes
+                    if s['name'] in AUTH_TYPE_MAPS
                 )
+            except StopIteration:
+                # If no auth scheme was specifically requested and an
+                # authSchemes list is present in the Endpoint object but none
+                # of the entries are supported, raise an exception.
+                fixable_with_crt = False
+                auth_type_options = [s['name'] for s in auth_schemes]
+                if not HAS_CRT:
+                    fixable_with_crt = any(
+                        scheme in CRT_SUPPORTED_AUTH_TYPES
+                        for scheme in auth_type_options
+                    )
```

Comment:
Similar suggestion here
```suggestion
for scheme in auth_schemes:
    if scheme['name'] in AUTH_TYPE_MAPS:
        name, scheme = scheme['name'], scheme
        break
else:
    fixable_with_crt = False
    auth_type_options. = [s['name' for s in auth_schemes]
    if not HAS_CRT:
        fixable_with_crt = any(
            scheme in CRT_SUPPORTED_AUTH_TYPES
            for scheme in auth_type_options
        )